pacific_rim_library/indexer.py

```diff
@@ -545,16 +543,32 @@ def download_thumbnail(self, prl_solr_document: PRLSolrDocument):
                     response = requests.get(original_thumbnail_url, timeout=30, stream=True)
                     # Fail on 4xx or 5xx
                     response.raise_for_status()
-                    # Make sure the Content-Type is what we expect. Some servers discriminate against robots.
-                    if re.match(re.compile('image/.+'), response.headers.get('Content-Type')):
+                    # Make sure the Content-Type is what we expect and that the server doesn't disallow robots
+                    response_content_type = response.headers.get('Content-Type')
+                    if re.match(re.compile('image/.+'), response_content_type):
                         with open(filepath, 'wb') as image_file:
                             for chunk in response.iter_content(chunk_size=1024):
                                 image_file.write(chunk)
                         logging.debug(
                             '%s thumbnail put on local filesystem at %s',
                             thumbnail_s3_key,
                             filepath)
-                        return filepath
+
+                        if not prl_solr_document.has_thumbnail_format():
+                            # Determine the format and rename the image file to use the newly-determined filetype ext
+                            prl_solr_document.set_thumbnail_format(response_content_type)
+                            new_filepath = os.path.join(
+                                os.path.abspath(os.path.expanduser(self.config['s3']['sync']['source'])),
+                                prl_solr_document.get_thumbnail_s3_key()
```

Comment:
The return value of `get_thumbnail_s3_key` will change if `has_thumbnail_format() == False` before a call to `set_thumbnail_format`. The result is a new URL for the thumbnail (and therefore, a new local file path that we should move the thumbnail to so that it matches the path part of the new URL).